pkg/statdb/statdb_test.go

```diff
@@ -5,461 +5,477 @@ package statdb_test
-		AuditSuccess:       true,
-		UpdateUptime:       false,
-	}
-	updateBatchReq := &pb.UpdateBatchRequest{
-		NodeList: []*pb.Node{node1, node2},
-	}
-	_, err = statdb.UpdateBatch(ctx, updateBatchReq)
-	assert.NoError(t, err)
-}
-
-func TestUpdateBatchEmpty(t *testing.T) {
-	dbPath := getDBPath()
-	statdb, db, err := getServerAndDB(dbPath)
-	assert.NoError(t, err)
-
-	nodeID1 := teststorj.NodeIDFromString("testnodeid1")
-
-	auditSuccessCount1, totalAuditCount1, auditRatio1 := getRatio(4, 10)
-	uptimeSuccessCount1, totalUptimeCount1, uptimeRatio1 := getRatio(8, 25)
-	err = createNode(ctx, db, nodeID1, auditSuccessCount1, totalAuditCount1, auditRatio1,
-		uptimeSuccessCount1, totalUptimeCount1, uptimeRatio1)
-	assert.NoError(t, err)
-
-	updateBatchReq := &pb.UpdateBatchRequest{
-		NodeList: []*pb.Node{},
-	}
-	resp, err := statdb.UpdateBatch(ctx, updateBatchReq)
-	assert.NoError(t, err)
-	assert.Equal(t, len(resp.StatsList), 0)
-}
-
-func TestCreateEntryIfNotExists(t *testing.T) {
-	dbPath := getDBPath()
-	statdb, db, err := getServerAndDB(dbPath)
-	assert.NoError(t, err)
-
-	nodeID1 := teststorj.NodeIDFromString("testnodeid1")
-	nodeID2 := teststorj.NodeIDFromString("testnodeid2")
-
-	auditSuccessCount1, totalAuditCount1, auditRatio1 := getRatio(4, 10)
-	uptimeSuccessCount1, totalUptimeCount1, uptimeRatio1 := getRatio(8, 25)
-	err = createNode(ctx, db, nodeID1, auditSuccessCount1, totalAuditCount1, auditRatio1,
-		uptimeSuccessCount1, totalUptimeCount1, uptimeRatio1)
-	assert.NoError(t, err)
-
-	node1 := &pb.Node{Id: nodeID1}
-	createIfNotExistsReq1 := &pb.CreateEntryIfNotExistsRequest{
-		Node: node1,
-	}
-	_, err = statdb.CreateEntryIfNotExists(ctx, createIfNotExistsReq1)
-	assert.NoError(t, err)
-
-	nodeInfo1, err := db.Get_Node_By_Id(ctx, dbx.Node_Id(nodeID1.Bytes()))
-	assert.NoError(t, err)
-	assert.EqualValues(t, nodeID1.Bytes(), nodeInfo1.Id)
-	assert.EqualValues(t, auditRatio1, nodeInfo1.AuditSuccessRatio)
-	assert.EqualValues(t, uptimeRatio1, nodeInfo1.UptimeRatio)
-
-	node2 := &pb.Node{Id: nodeID2}
-	createIfNotExistsReq2 := &pb.CreateEntryIfNotExistsRequest{
-		Node: node2,
-	}
-	_, err = statdb.CreateEntryIfNotExists(ctx, createIfNotExistsReq2)
-	assert.NoError(t, err)
-
-	nodeInfo2, err := db.Get_Node_By_Id(ctx, dbx.Node_Id(nodeID2.Bytes()))
-	assert.NoError(t, err)
-	assert.EqualValues(t, nodeID2.Bytes(), nodeInfo2.Id)
-	assert.EqualValues(t, 0, nodeInfo2.AuditSuccessRatio)
-	assert.EqualValues(t, 0, nodeInfo2.UptimeRatio)
-}
-
-func getDBPath() string {
-	return fmt.Sprintf("file:memdb%d?mode=memory&cache=shared", rand.Int63())
-}
-
-func getServerAndDB(path string) (sdb *statdb.StatDB, db *dbx.DB, err error) {
-	sdb, err = statdb.NewStatDB("sqlite3", path, zap.NewNop())
-	if err != nil {
-		return &statdb.StatDB{}, &dbx.DB{}, err
-	}
-	db, err = dbx.Open("sqlite3", path)
-	if err != nil {
-		return &statdb.StatDB{}, &dbx.DB{}, err
-	}
-	return sdb, db, err
-}
+	t.Run("TestCreateDoesNotExist", func(t *testing.T) {
```

Comment:
Let's not use subtests to keep test output cleaner.

Reply:
Tired to keep the changes to the minimal. Moby said he will work on it when he is refactoring statdb